### PR TITLE
Add initial site checks

### PR DESCRIPTION
**Pull request by zach-bob**
[Per this card](https://github.com/mojotech/situation-room/projects/1#card-82057978) and contrary to this branch's name - this PR implements the first pass for site_checks in relation to site. Please do not merge this PR until [PR 36](https://github.com/mojotech/situation-room/pull/36) is merged.

### CODE REVIEW

File: lib/situation_room/sites/site_controller.ex [hunk 1]
```diff
@@ -66,6 +66,8 @@ defmodule SituationRoom.Site.Controller do
     end
   end
 
+  forward("/:id/checks", to: SituationRoom.Site.Check.Controller)
```
Comment (cpjolicoeur) [inline lib/situation_room/sites/site_controller.ex:69]: I think this change should be included in the commit where the Controller is added.  Or the adding of the controller should move forward to this commit here.  There is not value in having the controller introduced but not consumed.  That breaks the atomic nature a bit.

File: lib/situation_room/site_checks/site_check_controller.ex [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  """
+  use Plug.Router
+  alias SituationRoom.Site.Check
+
+  plug(:match)
+
+  plug(Plug.Parsers,
+    parsers: [:urlencoded, :multipart],
+    pass: ["text/*"]
+  )
```
Comment (cpjolicoeur) [inline lib/situation_room/site_checks/site_check_controller.ex:13]: What are these parsers here for?  We don't need any parsing as we only have a GET endpoint?

File: priv/repo/migrations/20220513131026_create_pings.exs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+defmodule SituationRoom.Repo.Migrations.SituationRoom.Site.Check do
+  use Ecto.Migration
+
+  def change do
+    create table(:site_checks) do
+      add :site_id, references(:sites)
```
Comment (cpjolicoeur) [inline priv/repo/migrations/20220513131026_create_pings.exs:6]: This also cannot be null

**Review by cpjolicoeur — CHANGES_REQUESTED** [review]

File: lib/situation_room/site_checks/site_check.ex [hunk 4]
```diff
@@ -2,7 +2,7 @@ defmodule SituationRoom.Site.Check do
   @moduledoc """
   Documentation for `SituationRoom.Ping`.
   """
-  @derive {Jason.Encoder, only: [:id, :site_id, :status_code, :response_time]}
+  @derive {Jason.Encoder, except: [:__meta__]}
```
[the remark this answers is not in the document]
Comment (zach-bob) [reply lib/situation_room/site_checks/site_check.ex]: I thought that as well, but but when using `@derive Jason.Encoder`, I observe this error:
```
[error] #PID<0.474.0> running SituationRoom.Router (connection #PID<0.473.0>, stream id 1) terminated
Server: localhost:4001 (http)
Request: GET /sites/
** (exit) an exception was raised:
    ** (RuntimeError) cannot encode metadata from the :__meta__ field for SituationRoom.Site to JSON. This metadata is used internally by Ecto and should never be exposed externally.

You can either map the schemas to remove the :__meta__ field before encoding to JSON, or explicit list the JSON fields in your schema:

    defmodule SituationRoom.Site do
      # ...

      @derive {Jason.Encoder, only: [:name, :title, ...]}
      schema ... do
```

**Review by cpjolicoeur — APPROVED** [review]

**Pull request merged**